### ISSUE TITLE
`response.ok` is always true

### ISSUE THREAD

**Issue by lmachens** [ajaishankar/openapi-typescript-fetch#22]
👋 thank you for this great module.

I noticed, that an `APIError` is thrown if `response.ok` is falsy:

https://github.com/ajaishankar/openapi-typescript-fetch/blob/b62ee714e645c14b8e07c98aaaa53880e7d6c847/src/fetcher.ts#L163-L167

`ok` is omitted in `ApiError` constructor:
https://github.com/ajaishankar/openapi-typescript-fetch/blob/b62ee714e645c14b8e07c98aaaa53880e7d6c847/src/types.ts#L162

This means, that you always have `true` for `ok`, because it is not part of the error response.

It's not a real issue, but I expected that I could check for `ok` by myself to separate abnormal HTTP-statuses (4xx, 5xx) from network problems.
I suggest removing `ok` from the result, or do not throw an error if the response is not `ok`.

**Comment by ajaishankar** [comment]
@lmachens ApiError does not have an ok field, so won't be true.

```ts
export class ApiError extends Error {
  readonly headers: Headers
  readonly url: string
  readonly status: number
  readonly statusText: string
  readonly data: any

  constructor(response: Omit<ApiResponse, 'ok'>) {
    super(response.statusText)
    Object.setPrototypeOf(this, new.target.prototype)

    this.headers = response.headers
    this.url = response.url
    this.status = response.status
    this.statusText = response.statusText
    this.data = response.data
  }
}

```

Also status is included in the response/error (4xx, 5xx)

Plus non ok responses are strongly typed as documented [here](https://github.com/ajaishankar/openapi-typescript-fetch#typed-error-handling)

**Comment by simensol** [comment]
It is not entirely clear to me, where in the [Typed Error Handling](https://github.com/ajaishankar/openapi-typescript-fetch#typed-error-handling):

```typescript
const findPetsByStatus = fetcher.path('/pet/findByStatus').method('get').create()
const addPet = fetcher.path('/pet').method('post').create()

try {
  await findPetsByStatus({ ... })
  await addPet({ ... })
} catch(e) {
  // check which operation threw the exception
  if (e instanceof addPet.Error) {
    // get discriminated union { status, data } 
    const error = e.getActualType()
    if (error.status === 400) {
      error.data.validationErrors // only available for a 400 response
    } else if (error.status === 500) {
      error.data.errorMessage // only available for a 500 response
    } else {
      ...
    }
  }
}
```

can I detect if it is a network error (`ok=false`)?
